flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordOrEventCollectingResultPartitionWriter.java

```diff
@@ -48,7 +50,16 @@ public RecordOrEventCollectingResultPartitionWriter(
 
     @Override
     public void broadcastEvent(AbstractEvent event, boolean isPriorityEvent) throws IOException {
-        output.add(event);
+        // Go through the serialization/deserialization to provide better test coverage in the
+        // unit tests using test harnesses. Otherwise bugs in event serialisation would be only
+        // visible in ITCases or end to end tests.
+        try (BufferConsumer eventBufferConsumer =
+                EventSerializer.toBufferConsumer(event, isPriorityEvent)) {
+            AbstractEvent deserializedEvent =
+                    EventSerializer.fromBuffer(
+                            eventBufferConsumer.build(), getClass().getClassLoader());
```

Comment:
I guess `recycleBuffer` isn't mandatory here since it's on-heap buffer, but wouldn't heart.

Reply:
Yes, good catch, I've forgotten about it.